File: base_java/src/main/java/com/hesheng1024/base_java/BaseApplication.java
package com.hesheng1024.base_java;

import android.app.Application;
import android.content.Context;
import android.util.Log;

/**
 * @author hesheng1024
 * @date 2019/12/8 16:11:14
 */
public class BaseApplication extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        long startTime = System.currentTimeMillis();
        Log.e("startTime", String.valueOf(startTime));
    }

    @Override
    public void onCreate() {
        super.onCreate();
        ContextHolder.init(this, getBaseContext());
    }

    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        // Glide.get(this).onTrimMemory(level);
    }
}

File: base_kotlin/src/main/java/com/hesheng1024/base_kotlin/BaseApplication.kt
package com.hesheng1024.base

import android.app.Application
import android.content.Context
import android.util.Log
import com.hesheng1024.base.utils.ContextHolder

/**
 *
 * @author hesheng1024
 * @date 2020/2/7 10:38
 */
abstract class BaseApplication : Application() {

    companion object {
        private const val TAG = "BackupApplication"
    }


    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        Log.i(TAG, "startTime: ${System.currentTimeMillis()}")
    }

    override fun onCreate() {
        super.onCreate()
        ContextHolder.holdMainApplication(this)
        ContextHolder.holdMainContext(baseContext)
    }
}